.gitignore

```diff
@@ -0,0 +1,15 @@
+# Compiled files
+*.o
+*.so
+*.rlib
+*.dll
+
+# Executables
+*.exe
+
+# Generated by Cargo
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here http://doc.crates.io/guide.html#cargotoml-vs-cargolock
+Cargo.lock
```

Comment:
I'd prefer just
```
/target
Cargo.lock
```
These two don't require explanation, the rest are not relevant.